### PR TITLE
#51 leaving session by using api

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #51 
Implemented some of the issues and left puzzles for the rest

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#56](https://github.com/ammaratef45/attendance/issues/56) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) everybody who has role `REV` is banned at [#56](https://github.com/ammaratef45/attendance/issues/56); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
This pull request [#56](https://github.com/ammaratef45/attendance/issues/56) is assigned to @bahaaib[/z](https://www.0crat.com/u/bahaaib), here is [why](https://www.0crat.com/footprint/CH951175M/9fe92c4a-8e42-4648-9ee3-325c84d22122); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @ammaratef45[/z](https://www.0crat.com/u/ammaratef45) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; @bahaaib[/z](https://www.0crat.com/u/bahaaib) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

File: attendance/lib/offline_page/offline_page_viewmodel.dart [hunk 1]
```diff
@@ -55,7 +55,7 @@ abstract class OfflinePageViewModel extends State<OfflinePage> {
   /// scan a QR code.
   Future<void> scan() async {
     try {
-      String barcode = await BarcodeScanner.scan();
+      final String barcode = await BarcodeScanner.scan();
       Session session = Session.fromMap(json.decode(barcode));
       DateTime now = new DateTime.now();
```
Comment (Bahaaib) [inline attendance/lib/offline_page/offline_page_viewmodel.dart]: @ammaratef45 Please get rid of the 'new'

**Comment by ammaratef45** [comment]
@Bahaaib done please check

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31):

**Comment by 0crat** [comment]
Job was finished in 8 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#56](https://github.com/ammaratef45/attendance/issues/56) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Pull request merged**